packages/suite/src/hooks/wallet/coinmarket/form/useCoinmarketExchangeForm.ts

```diff
@@ -356,18 +368,28 @@ export const useCoinmarketExchangeForm = ({
             if (!trade) {
                 trade = selectedQuote;
             }
-            if (!trade || !refundAddress) return false;
+            if (!quotesRequest || !trade || !refundAddress || !trade.quoteId) return false;
 
             if (trade.isDex && !trade.fromAddress) {
                 trade = { ...trade, fromAddress: refundAddress };
             }
 
             setCallInProgress(true);
+            dispatch(coinmarketExchangeActions.saveTransactionId(undefined));
+
+            const returnUrl = await createQuoteLink(
+                quotesRequest,
+                account,
+                { selectedFee: selectedFeeRecomposedAndSigned, composed },
+                trade.quoteId!,
```

Comment:
It should not be undefined, can you delete `!`?

Reply:
fixed in https://github.com/trezor/trezor-suite/pull/15612/commits/1a7856840161de71d77e77f71e36195669db9168